### PR TITLE
Openblas clang fix

### PR DESCRIPTION
**Pull request by bvanessen**
As of version 0.2.20 OpenBLAS will now compile with clang.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Should we check whether or not it is Apple clang? I guess it'll fail when it tries to look up `self.compiler.openmp_flag`.

**Comment by bvanessen** [comment]
I don't know.  I was working on a linux based system for this.

File: var/spack/repos/builtin/packages/openblas/package.py [hunk 1]
```diff
@@ -84,7 +84,8 @@ def check_compilers(self):
                 'OpenBLAS requires both C and Fortran compilers!'
             )
         # Add support for OpenMP
-        if '+openmp' in self.spec and self.spec.satisfies('%clang'):
+        if (('+openmp' in self.spec) and self.spec.satisfies('%clang') and
+            (('@:0.2.19' in self.spec) or ('platform=darwin' in self.spec))):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openblas/package.py]: Technically, I think people can build `llvm` on macOS and it should support OpenMP just fine. A better check, which `qt` uses, would be:
```python
str(self.spec.compiler.version).endswith('-apple')
```
You should also update the comment below.

**Comment by alalazo** [comment]
Failing on `flake8`:
```console
=======================================================
flake8: running flake8 code checks on spack.
Modified files:
  var/spack/repos/builtin/packages/openblas/package.py
=======================================================
var/spack/repos/builtin/packages/openblas/package.py:98: [E303] too many blank lines (2)
```

**Pull request merged**